src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1908,16 +1908,22 @@ export default class CanvasController {
         cc.translate(CanvasStageModel._panX, CanvasStageModel._panY);
         cc.beginPath();
 
-        for (let i = 0; i < airportModel.maps.base.length; i++) {
-            const mapItem = airportModel.maps.base[i];
-            cc.moveTo(
-                CanvasStageModel.translateKilometersToPixels(mapItem[0]),
-                -CanvasStageModel.translateKilometersToPixels(mapItem[1])
-            );
-            cc.lineTo(
-                CanvasStageModel.translateKilometersToPixels(mapItem[2]),
-                -CanvasStageModel.translateKilometersToPixels(mapItem[3])
-            );
+        for (let map of airportModel.maps) {
+            if (!map || !map.lines.length || map.hidden) {
+                continue;
+            }
+            const lines = map.lines;
```

Comment:
you could use destructuring here:
```js
const { lines } = map;
```

Reply:
Thanks for pointing that out. I'd spotted the usage elsewhere and hadn't looked into the usage and reasoning.